docs/src/examples/ITensor.md

```diff
@@ -62,6 +62,36 @@ el = T[j=>1,i=>2,k=>3]
 println("The (i,j,k) = (2,1,3) element of T is ",el)
 ```
 
+## Constructing ITensors from Arrays
+
+To initialize all of the elements of an ITensor at once, you
+can pass a Julia array into the ITensor constructor.
+
+For example, if we want to construct an ITensor `A` with indices 
+`i,j` we can initialize it from a matrix as follows:
+
+```julia
+M = [1.0 2.0;
+     3.0 4.0]
+
+i = Index(2,"i")
+j = Index(2,"j")
+
+A = ITensor(M,i,j)
+```
+
+More generally we can use an nth-order (n-dimensional) Julia array to initialize an ITensor:
+
+```julia
+T = randn(4,7,2)
+
+k = Index(4,"index_k")
+l = Index(7,"index_l")
+m = Index(2,"index_m")
+
+B = ITensor(T,k,l,m)
+```
```

Comment:
We could also mention ways to make this more compact, like:
```julia
k, l, m = Index.(size(T), ("index_k", "index_l", "index_m"))
```
That can give people ideas for how to use Julia's built in tools to make things easier. We could put "advanced" tips like that in collapsible blocks that you have to click on to see: https://discourse.julialang.org/t/documenter-jl-and-collapsible-markdown/50898

Reply:
I didn't know that clickable dropdowns were part of html now. That's great! On itensor.org I had been using a more complicated JavaScript trick for that. I wasn't quite able to get it to work like in that example, though. Even when I wrapped the html in a ```@raw html ... ``` block that Documenter uses for raw code, it sort of worked but when I put another code block inside, it got confused and pulled the entire rest of the page's contents into the dropdown.

I was thinking we should have some examples and tips about indices. We could put them in the ITensor examples, for the case of dense indices, and have a separate page all about QN Indices and QN ITensor code examples. (Indices by themselves may not warrant an entire section is what I'm thinking, though maybe they do in order to discuss things like id's and tags and so on.)

Reply:
Another neat way to make a group of indices:
```julia
is = Index.(size(T), ("i$n" for n in 1:ndims(T)))
# is = [Index(size(T, n), "i$n") for n in 1:ndims(T)]
ITensor(T, is...)
```
since I think people find making a big group of indices tedious but might not realize how to use Julia tools to help with it. I find constructors like the above make writing random tests of contraction much easier, for example:
```julia
using Random
i = [Index(n, "i$n") for n in 1:12]
A = randomITensor(shuffle(i[1:7])...)
B = randomITensor(shuffle(i[4:end])...)
A * B
```

Also, should we mention the `itensor` constructor here as well? That could also be in a collapsible "advanced" section.

Reply:
I do agree these are neat and we should put them somewhere, but I do think it should be in a more advanced sort of section. A dropdown at the end of a closely related section is one good idea. We could also have a separate page or pages called "Advanced ITensor Examples" etc. Some of these techniques are most useful for things like writing tests, whereas a lot of code "in the wild" may only involve making a small handful of indices or using something like `siteinds` though it definitely varies from application to application.

Reply:
I keep being amazed what you can do with broadcasting in Julia. It's not something I've been used to since I've mostly used C++ before!

Reply:
That's strange, I just tested pasting this into a section of the docs and it works correctly for me:
``````julia
```@raw html
<details><summary>Advanced</summary>
```

```julia
print("hello world!")
```

```@raw html
</details>
```
``````

Reply:
Great, I see. You did it differently from me, by having two @raw blocks. I was trying to use just one that wrapped around all of the contents inside, including the code block.